pyproject.toml

```diff
@@ -49,22 +48,52 @@ dependencies = [
 ]
 dynamic = ["version"]
 
-[tool.setuptools.packages.find]
-exclude = ["*.pyc", ".pytest_cache/*", ".hypothesis/*"]
+[project.optional-dependencies]
+doc = [
+    "jinja2==3.0.0",
+    "sphinx-rtd-theme==2.0.0",
+    "Sphinx",
+    "breathe",
+]
+test = [
+    "pytest",
+    "hypothesis",
+    "psutil",
+    "pyarrow",
+]
 
 [project.urls]
 homepage = "https://github.com/TileDB-Inc/TileDB-Py"
 
 [tool.setuptools_scm]
-version_scheme = "guess-next-dev"
-local_scheme = "no-local-version"
 version_file = "tiledb/_generated_version.py"
 
+[tool.scikit-build]
+wheel.expand-macos-universal-tags = true
+metadata.version.provider = "scikit_build_core.metadata.setuptools_scm"
+wheel.packages = ["tiledb"]
+wheel.exclude = [
+    "tiledb/cc/",
+    "tiledb/*.cc",
+    "tiledb/*.cmake",
+    "tiledb/*.pyx",
+    "tiledb/*.pxd",
+    "tiledb/*.pxi",
+]
+
+[tool.scikit-build.cmake.define]
+TILEDB_PATH = {env="TILEDB_PATH"}
+TILEDB_VERSION = {env="TILEDB_VERSION"}
```

Comment:
Let's also add an option to choose SHA from terminal.

Reply:
Done